### PR TITLE
Rewrite gradle in Kotlin

### PR DESCRIPTION
**Pull request by 5HT2**

### CODE REVIEW

**Comment by 5HT2** [comment]
I'm not sure how to run tests since the changes done in master, I have no idea if this works or not. I updated it with gradle changes from master

**Comment by 5HT2** [comment]
image
Actually, I'm blind. Here you go

**Review by eiknat — APPROVED** [review]

**Comment by eiknat** [comment]
lgtm!

File: src/main/kotlin/lemmy/client/Library.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
```
Comment (5HT2) [inline src/main/kotlin/lemmy/client/Library.kt:1]: I'm not sure if we should leave these, or just rely on our own tests?

File: src/test/kotlin/lemmy/client/LibraryTest.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*
```
Comment (5HT2) [inline src/test/kotlin/lemmy/client/LibraryTest.kt:1]: Above comment

File: gradle.properties [hunk 3]
```diff
@@ -1 +0,0 @@
-kotlin.code.style=official
```
Comment (5HT2) [inline gradle.properties:1]: this is set by default in newer Kotlin plugins / Kotlin Gradle iirc

**Comment by 5HT2** [comment]
*Now* this should be ready for review. Sorry about the delay

File: src/main/kotlin/lemmy/client/Library.kt (hunk 1 above)
Comment (eiknat) [reply src/main/kotlin/lemmy/client/Library.kt:1]: can probably remove. the tests that are there now should be fine. i need to figure out github actions, never used it before ^_^

**Comment by 5HT2** [comment]
I'll leave you to decide what we do about the automatically-added Kotlin Gradle tests

**Pull request merged**